spring-integration-core/src/main/java/org/springframework/integration/graph/IntegrationNode.java

```diff
@@ -76,6 +81,27 @@ public Stats getStats() {
 		return this.properties.size() == 0 ? null : this.properties;
```

Comment:
`@Nullable` on this getter? Or, maybe we should just always return it, even if empty.

Also, now that we have a mechanism to modify the map, maybe we should return an `UnmodifableMap` - if so, it should be a field instead of wrapping the map on every call.

Reply:
Do you mean we should use a `BiConsumer<NamedComponent, Map<String, Object>>`  instead of that `Function` for additional properties?

Reply:
?

No; I just mean that now we have `IN.addProperty()` and `addProperties()` we no longer need to expose a direct reference to properties field.